packages/core/src/services/createSignedURLService.ts

```diff
@@ -6,27 +6,29 @@ import {
 } from '@unchainedshop/core-files';
 
 export type CreateSignedURLService = (
-  params: { directoryName: string; fileName: string; meta?: any },
-  unchainedAPI: { modules: { files: FilesModule } },
+  params: { directoryName: string; fileName: string; meta?: any; isPrivate?: boolean },
+  unchainedAPI: { userId?: string; modules: { files: FilesModule } },
 ) => Promise<SignedFileUpload>;
 
 export const createSignedURLService: CreateSignedURLService = async (
-  { directoryName, fileName, meta },
+  { directoryName, fileName, meta, isPrivate },
   unchainedContext,
 ) => {
   const {
     modules: { files },
   } = unchainedContext;
   const fileUploadAdapter = getFileAdapter();
-  const preparedFileData = await fileUploadAdapter.createSignedURL(
-    directoryName,
-    fileName,
-    unchainedContext,
-  );
-  const fileData = getFileFromFileData(preparedFileData, meta);
+  if (isPrivate && fileUploadAdapter.key !== 'shop.unchained.file-upload-plugin.gridfs')
```

Comment:
🫣 what is the idea of adapters when the caller adds a conditional for a specific adapter?